### PR TITLE
EVG-19853: Log the metric name, the operation and the actor right before the metric starts being collected

### PR DESCRIPTION
**Pull request by PyGeek03**
**Jira Ticket:** EVG-19853

Note: Matt Boros and Hana Pearlman only review if the UX of this change is what these users expect. Performance Tooling team owns this change, and you should contact our team if this change causes any unexpected issues.

**Whats Changed:**  
Log the metric name, the operation and the actor right before the metric starts being collected. This logging is done at TRACE log level to avoid creating extra logging computation load and noise when running on Evergreen.

For example, here's part of the log output for [TimeSeriesSort.yml](https://github.com/mongodb/genny/blob/f014c99eb7681e7b337b12e586469137e3a2343f/src/workloads/query/TimeSeriesSort.yml) (the full output can be found [here](https://privatebin.corp.mongodb.com/?8a03d69bf458cf9a#EncCFkCs8u5P21VxFpAxY8D85NUYRma1uc7SJDSPhQrE)):

```
> ./run-genny workload -v trace src/workloads/query/TimeSeriesSort.yml
...
[2023-06-27 18:25:02.832951] [0x00000001e8a9de00] [trace]   Started collecting metric `canary_TimeSeriesSort.Setup` for operation `Setup` of actor `TimeSeriesSort`
[2023-06-27 18:25:02.834596] [0x000000017d5b7000] [trace]   Started collecting metric `canary_TimeSeriesSort.ActorStarted` for operation `ActorStarted` of actor `TimeSeriesSort`
[2023-06-27 18:25:02.834618] [0x000000017d7e7000] [trace]   Started collecting metric `canary_TimeSeriesSort.ActorStarted` for operation `ActorStarted` of actor `TimeSeriesSort`
[2023-06-27 18:25:02.834639] [0x000000017d643000] [trace]   Started collecting metric `canary_TimeSeriesSort.ActorStarted` for operation `ActorStarted` of actor `TimeSeriesSort`
[2023-06-27 18:25:02.834649] [0x000000017d75b000] [trace]   Started collecting metric `canary_TimeSeriesSort.ActorStarted` for operation `ActorStarted` of actor `TimeSeriesSort`
[2023-06-27 18:25:02.834659] [0x000000017d873000] [trace]   Started collecting metric `canary_TimeSeriesSort.ActorStarted` for operation `ActorStarted` of actor `TimeSeriesSort`
[2023-06-27 18:25:02.834667] [0x000000017d6cf000] [trace]   Started collecting metric `canary_TimeSeriesSort.ActorStarted` for operation `ActorStarted` of actor `TimeSeriesSort`
[2023-06-27 18:25:02.834689] [0x000000017d6cf000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":0,"request":1}
') start
[2023-06-27 18:25:02.834697] [0x000000017d6cf000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":0,"request":1}
') end
[2023-06-27 18:25:02.834704] [0x000000017d6cf000] [info]    Beginning phase 0
[2023-06-27 18:25:02.834727] [0x000000017d6cf000] [trace]   Started collecting metric `CreateTimeSeriesCollection.CreateTimeSeriesCollection` for operation `CreateTimeSeriesCollection` of actor `CreateTimeSeriesCollection`
[2023-06-27 18:25:02.834748] [0x000000017d5b7000] [trace]   Started collecting metric `canary_PhaseTimingRecorder.Phase` for operation `Phase` of actor `PhaseTimingRecorder`
[2023-06-27 18:25:02.903411] [0x000000017d6cf000] [info]    Ended phase 0
[2023-06-27 18:25:02.903444] [0x000000017d6cf000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Ended phase","phase":0,"request":1}
') start
[2023-06-27 18:25:02.903453] [0x000000017d6cf000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Ended phase","phase":0,"request":1}
') end
[2023-06-27 18:25:02.903490] [0x000000017d5b7000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":1,"request":1}
') start
[2023-06-27 18:25:02.903498] [0x000000017d5b7000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":1,"request":1}
') end
[2023-06-27 18:25:02.903505] [0x000000017d5b7000] [info]    Beginning phase 1
[2023-06-27 18:25:02.903519] [0x000000017d5b7000] [trace]   Started collecting metric `canary_PhaseTimingRecorder.Phase` for operation `Phase` of actor `PhaseTimingRecorder`
[2023-06-27 18:25:02.903538] [0x000000017d6cf000] [trace]   Started collecting metric `CreateTimeSeriesCollection.DropAllIndexesUponCreation` for operation `DropAllIndexesUponCreation` of actor `CreateTimeSeriesCollection`
[2023-06-27 18:25:02.905276] [0x000000017d75b000] [info]    Starting to insert: 100000 docs into Collection0
[2023-06-27 18:25:02.905292] [0x000000017d75b000] [trace]   Started collecting metric `InsertData.TotalBulkInsert` for operation `TotalBulkInsert` of actor `InsertData`
[2023-06-27 18:25:02.920782] [0x000000017d75b000] [trace]   Started collecting metric `InsertData.IndividualBulkInsert` for operation `IndividualBulkInsert` of actor `InsertData`
[2023-06-27 18:25:03.089979] [0x000000017d75b000] [trace]   Started collecting metric `InsertData.IndividualBulkInsert` for operation `IndividualBulkInsert` of actor `InsertData`
[2023-06-27 18:25:03.261141] [0x000000017d75b000] [trace]   Started collecting metric `InsertData.IndividualBulkInsert` for operation `IndividualBulkInsert` of actor `InsertData`
[2023-06-27 18:25:03.423618] [0x000000017d75b000] [trace]   Started collecting metric `InsertData.IndividualBulkInsert` for operation `IndividualBulkInsert` of actor `InsertData`
[2023-06-27 18:25:03.481578] [0x000000017d75b000] [info]    Done with load phase. All 100000 documents loaded into Collection0
[2023-06-27 18:25:03.481610] [0x000000017d75b000] [info]    Ended phase 1
[2023-06-27 18:25:03.481622] [0x000000017d75b000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Ended phase","phase":1,"request":1}
') start
[2023-06-27 18:25:03.481629] [0x000000017d75b000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Ended phase","phase":1,"request":1}
') end
[2023-06-27 18:25:03.481761] [0x000000017d6cf000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":2,"request":1}
') start
[2023-06-27 18:25:03.481786] [0x000000017d6cf000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":2,"request":1}
') end
[2023-06-27 18:25:03.481796] [0x000000017d6cf000] [info]    Beginning phase 2
[2023-06-27 18:25:03.481835] [0x000000017d643000] [trace]   Started collecting metric `Quiesce.Quiesce` for operation `Quiesce` of actor `Quiesce`
[2023-06-27 18:25:03.481844] [0x000000017d5b7000] [trace]   Started collecting metric `canary_PhaseTimingRecorder.Phase` for operation `Phase` of actor `PhaseTimingRecorder`
[2023-06-27 18:25:03.481851] [0x000000017d643000] [debug]   QuiesceActor quiescing cluster.
[2023-06-27 18:25:03.573553] [0x000000017d643000] [info]    Ended phase 2
[2023-06-27 18:25:03.573583] [0x000000017d643000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Ended phase","phase":2,"request":1}
') start
[2023-06-27 18:25:03.573592] [0x000000017d643000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Ended phase","phase":2,"request":1}
') end
[2023-06-27 18:25:03.573692] [0x000000017d873000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":3,"request":1}
') start
[2023-06-27 18:25:03.573718] [0x000000017d873000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":3,"request":1}
') end
[2023-06-27 18:25:03.573731] [0x000000017d873000] [info]    Beginning phase 3
[2023-06-27 18:25:03.573775] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:03.573794] [0x000000017d5b7000] [trace]   Started collecting metric `canary_PhaseTimingRecorder.Phase` for operation `Phase` of actor `PhaseTimingRecorder`
[2023-06-27 18:25:03.631730] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:03.688218] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:03.745284] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:03.807077] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:03.865817] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:03.924890] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:03.986682] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.045804] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.100739] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.158245] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.209044] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.269396] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.327080] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.383250] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.437267] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.485911] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.535651] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.589320] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.637819] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.686513] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.734896] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.787500] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.838883] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.888936] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.937290] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:04.986968] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.038376] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.085983] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.137245] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.189676] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.243232] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.294009] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.343923] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.392601] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.440849] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.491559] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.540344] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.588804] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.640010] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.693773] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.750895] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.808233] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.859963] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.910483] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:05.968277] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:06.019244] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:06.069187] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:06.119570] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:06.169218] [0x000000017d873000] [trace]   Started collecting metric `Queries.SortQueryBoundedSort` for operation `SortQueryBoundedSort` of actor `Queries`
[2023-06-27 18:25:06.217237] [0x000000017d873000] [info]    Ended phase 3
[2023-06-27 18:25:06.217390] [0x000000017d873000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Ended phase","phase":3,"request":1}
') start
[2023-06-27 18:25:06.217439] [0x000000017d873000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Ended phase","phase":3,"request":1}
') end
[2023-06-27 18:25:06.217614] [0x000000017d5b7000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":4,"request":1}
') start
[2023-06-27 18:25:06.217669] [0x000000017d5b7000] [debug]   ExternalPhaseCoordinator::onPhase('{ "message": "Beginning phase","phase":4,"request":1}
') end
[2023-06-27 18:25:06.217713] [0x000000017d5b7000] [info]    Beginning phase 4
...
```

**Patch testing results:**
[sys-perf](https://spruce.mongodb.com/version/649ba22ae3c331a1301675e5/tasks)

genny workloads (eg. [big_update_10k](https://spruce.mongodb.com/task/sys_perf_linux_3_node_replSet.2022_11_big_update_10k_patch_b7fecb4445d3e54a601e4e4dd79b9d57d082117f_649ba22ae3c331a1301675e5_23_06_28_03_00_46/logs?execution=0&sortBy=STATUS&sortDir=ASC)) don't log at TRACE log level, but [genny_canaries](https://spruce.mongodb.com/task/sys_perf_linux_standalone.2022_11_genny_canaries_patch_2ef8927c2b6731fb68ed09dcbc80d0fd4c384ab0_64994de7e3c331b67ab443ab_23_06_26_08_36_43?execution=0) workload does. I'll look into why, but I think TRACE log level is low enough to prevent extra logging computation load and noise when running genny workloads on Evergreen.

### CODE REVIEW

File: src/metrics/include/metrics/operation.hpp [hunk 1]
```diff
@@ -255,10 +255,6 @@ class OperationImpl final : private boost::noncopyable {
     }
 
 private:
-    /*
-     * Actor count and phase number will be used in Poplar metrics. Right now they
-     * are unused.
-     */
```
Comment (PyGeek03) [inline src/metrics/include/metrics/operation.hpp:261]: I removed this comment because [the commit that introduce this comment doesn't actually add actor count](https://github.com/mongodb/genny/commit/c66113eaa8652f2375accbc24429a39da0910a75#diff-9951fc5c4b608ec55c6d5473fa06f5802e6b731eca2292a233fd8d46c3dd4365R224-R235), and phase number has been deleted in [a previous commit](https://github.com/mongodb/genny/commit/9e0031086b5227f41660a1344465ccb1a46bec1c#diff-9951fc5c4b608ec55c6d5473fa06f5802e6b731eca2292a233fd8d46c3dd4365L268).

File: src/metrics/include/metrics/operation.hpp [hunk 2]
```diff
@@ -387,6 +383,7 @@ class OperationT final {
     explicit OperationT(internals::OperationImpl<ClockSource>& op) : _op{std::addressof(op)} {}
 
     OperationContextT<ClockSource> start() {
+        BOOST_LOG_TRIVIAL(debug) << this->_op->getActorName() << " : " << this->_op->getOpName();
```
Comment (HanaPearlman) [inline src/metrics/include/metrics/operation.hpp:386]: [Optional] I was using the `Actor : OperationMetric` syntax as an example. I'm not sure if we are trying to keep each log line very succinct on purpose, but I wonder if it would be clearer to write out something a little more verbose like `Started collecting metric {metric} for actor {actor}` just so it's very obvious what the log line means.

**Review by HanaPearlman — APPROVED** [review]
LGTM, with one optional comment!

**Review by mattBoros — APPROVED** [review]
LGTM, you might want a reviewer from perf tooling as well.

Also I'm wondering if this would break any scripts that parse output from Genny? My guess is probably not since this only applies when debug log level is on.

**Review by thessem — COMMENTED** [review]
I am currently a little concerned by the logging overhead in a potentially sensitive part of the system. I would like to see a reasoned statement that this wont make tests noisier in this PR. 

Additionally, I think @rtimmons's comment on [Jira](https://jira.mongodb.org/browse/EVG-19853?focusedCommentId=5511776&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-5511776) is worth explicitly addressing, given the above paragraph.

Given these are actual questions and not me trying to be socratic, this is not a request for changes or an approval

File: src/metrics/include/metrics/operation.hpp (hunk 2 above)
Comment (PyGeek03) [reply src/metrics/include/metrics/operation.hpp]: Oh I actually thought you prefer to keep it succinct. Yes I prefer the more obvious version as well, and have pushed a change for that.

Comment (PyGeek03) [reply src/metrics/include/metrics/operation.hpp]: I've just taken a look at this again. Turns out adding metric name logging isn't complicated and shouldn't incur any computational cost, so I've added it as well.

Comment (PyGeek03) [reply src/metrics/include/metrics/operation.hpp]: Turns out my simple change to enable metric name logging actually breaks cmake-test. I've reverted that change, and will investigate why.

**Comment by PyGeek03** [comment]
> Also I'm wondering if this would break any scripts that parse output from Genny? My guess is probably not since this only applies when debug log level is on.

genny workloads (eg. [big_update_10k](https://spruce.mongodb.com/task/sys_perf_linux_3_node_replSet.2022_11_big_update_10k_patch_2ef8927c2b6731fb68ed09dcbc80d0fd4c384ab0_64994de7e3c331b67ab443ab_23_06_26_08_36_43?execution=0)) don't log at TRACE log level, but [genny_canaries](https://spruce.mongodb.com/task/sys_perf_linux_standalone.2022_11_genny_canaries_patch_2ef8927c2b6731fb68ed09dcbc80d0fd4c384ab0_64994de7e3c331b67ab443ab_23_06_26_08_36_43?execution=0) workload does. I'll look into why, but I think TRACE log level is low enough to prevent extra logging computation load and noise when running genny workloads on Evergreen.

File: src/metrics/include/metrics/operation.hpp [hunk 3]
```diff
@@ -387,6 +392,9 @@ class OperationT final {
     explicit OperationT(internals::OperationImpl<ClockSource>& op) : _op{std::addressof(op)} {}
 
     OperationContextT<ClockSource> start() {
+        BOOST_LOG_TRIVIAL(trace) << "Started collecting metric `" << this->_op->getMetricName()
+                                 << "` for operation `" << this->_op->getOpName() << "` of actor `"
+                                 << this->_op->getActorName() << "`";
```
Comment (rtimmons) [inline src/metrics/include/metrics/operation.hpp:397]: This looks like a lot of overhead for each metric invocation. Does BOOST_LOG_TRIVIAL effectively nop `operator<<` if TRACE isn't enabled? We'd still pay the "cost" of invoking `this->_op->getMetricName()` and the like, but I think we can assume those are trivial (please ensure they're `const` and return refs).

**Review by rtimmons — COMMENTED** [review]
Given this is in the critical-path, I'd like to see some sys-perf patch-builds proving this doesn't introduce latency / performance regressions.

I'm not sure why the cmake tests would care about this, but that feels like a bug in how we've configured cmake-test :)

As a totally different alternative:
Actors construct their `Operation`s at constructor-time. A dry-run workload should construct the same `metrics::Registry` as a production run. As an alternative to printing at runtime, there could be a member-function on the `Registry` that prints the metrics associated with each Phase, and this could be done in the [dry-run logic](https://github.com/mongodb/genny/blob/cfbd1c56e92498e938ada9d1a3aca0a618708d4c/src/driver/src/DefaultDriver.cpp#L220-L225).

E.g. add a `metrics::Registry::reportDryRunMetricsNames` or something that outputs a little text block of the metrics registered with each Phase.

```
Metrics To Collect:
  Phase 0:
    Loader.Insert
  Phase 1:
    Crud.0.FindOne
```

This would plainly point out if Metrics are being reported unexpectedly together indicating mis-configuration. E.g. 

```
Metrics To Collect:
  Phase 0:
    Loader.Insert
    Crud.0.FindOne
```

makes it pretty obvious that the FindOne is going to happen concurrently with the Loader which is probably not what the user wants. (Idk the best actual output format; just something that's easy to visually inspect.)

This may be slightly annoying, though, because we'd have to plumb through the Phase number to [the Operation factory functions](https://github.com/mongodb/genny/blob/cfbd1c56e92498e938ada9d1a3aca0a618708d4c/src/metrics/include/metrics/metrics.hpp#L202-L229), but I think that amount of plumbing and cleanup would be worthwhile. The metrics name book-keeping logic is v annoying (with various dots and defaults logic and logic in registry vs op vs context) and could certainly use some 💅 at some point. Perhaps plumbing through the Phase number would be good enough "training" to do a bit more surgery/💅 as a second pass.

And finally as a third option:
We've still got the CSV metrics output format if I remember correctly. I wonder if we could just print that at the end of the dry-run workloads and it'd show the metrics being collected during the dry-run. Perhaps a bit harder to visually parse given the naming doesn't always include the Phase number.

TLDR:
If we can safely say this adds no meaningful overhead *and* is easy to run and visually inspect to spot mis-configurations during workload/yaml development, then LGTM. But if it adds overhead or is too noisy to be useful during debugging or development (idk if that's true either way) then I think there are a few other, deeper options to consider.

File: src/metrics/include/metrics/operation.hpp (hunk 3 above)
Comment (PyGeek03) [reply src/metrics/include/metrics/operation.hpp:397]: Yes it is nop if log level is set higher than TRACE. According to [Boost.Log documentation](https://www.boost.org/doc/libs/1_82_0/libs/log/doc/html/log/tutorial/trivial_filtering.html), it "is only executed if the record passed filtering". Genny sets up filtering [here](https://github.com/mongodb/genny/blob/cfbd1c56e92498e938ada9d1a3aca0a618708d4c/src/driver/src/DefaultDriver.cpp#L159C1-L159C1), and [the default filter level is INFO](https://github.com/mongodb/genny/blob/cfbd1c56e92498e938ada9d1a3aca0a618708d4c/src/driver/src/DefaultDriver.cpp#L376-L378), which is higher than TRACE.

**Review by thessem — CHANGES_REQUESTED** [review]
We have discussed this internally and decided to try and achieve this using the dry run approach mentioned by Ryan above. 

I think the dry-run approach is closer to solving the underlying request here, given that the request seems to be more about develop time debugging than run-time observability. Needing to actually run the entire Genny workload to see if you got your YAML indentations right should be avoided.

If this were to go ahead in the method given here, I agree, I would like to see convincing proof that the logging doesn't change the performance profile. I have certainly accidentally destroyed the perf of a few programs both in the "It is slow to write text" side and accidentally trying to evaluate expensive things in no-op trace level metrics (since they're evaluated before the log gets dropped)

File: src/metrics/include/metrics/operation.hpp (hunk 3 above)
Comment (thessem) [reply src/metrics/include/metrics/operation.hpp:397]: Huh: "Remember that the streaming expression is only executed if the record passed filtering." well TIL

**Comment by PyGeek03** [comment]
I've used Perf Analyzer to test whether this PR introduces any performance regression, and the result is no:

https://performance-analyzer.server-tig.prod.corp.mongodb.com/perf-analyzer-viz/?comparison_id=58867636-c424-4ac4-9479-f2561fc2b2c1&percent_filter=0%7C%7C100&z_filter=0%7C%7C10
